service/discovery/azure/azure.go

```diff
@@ -205,3 +215,77 @@ func initClient[T any](existingClient *T, d *azureDiscovery, fun ClientCreateFun
 
 	return
 }
+
+// listPager loops all values from a [runtime.Pager] object from the Azure SDK and issues a callback for each item. It
+// takes the following arguments:
+//   - d, an [azureDiscovery] struct,
+//   - newListAllPager, a function that supplies a [runtime.Pager] listing all resources of a specific Azure client,
+//   - newListByResourceGroupPager, a function that supplies a [runtime.Pager] listing all resource of a specific resource group,
+//   - resToValues1, a function that takes the response from a single page of newListAllPager and returns its values,
```

Comment:
Why do you vall it resToValues1 and resToValues2 and not resAll and resByResourceGroup are similiar?